dm/master/workerrpc/interface.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workerrpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/dm/dm/pb"
+)
+
+// CmdType represents the concrete request type in Request or response type in Response.
+type CmdType uint16
+
+// CmdType values.
+const (
+	CmdStartSubTask CmdType = 1 + iota
+	CmdOperateSubTask
+	CmdUpdateSubTask
+
+	CmdQueryStatus
+	CmdQueryError
+	CmdQueryTaskOperation
+	CmdQueryWorkerConfig
+
+	CmdHandleSubTaskSQLs
+	CmdExecDDL
+	CmdBreakDDLLock
+
+	CmdSwitchRelayMaster
+	CmdOperateRelay
+	CmdPurgeRelay
+	CmdUpdateRelay
+	CmdMigrateRelay
+
+	CmdFetchDDLInfo
+)
+
+// Request wraps all dm-worker rpc requests.
+type Request struct {
+	Type CmdType
+
+	StartSubTask   *pb.StartSubTaskRequest
+	OperateSubTask *pb.OperateSubTaskRequest
+	UpdateSubTask  *pb.UpdateSubTaskRequest
+
+	QueryStatus        *pb.QueryStatusRequest
+	QueryError         *pb.QueryErrorRequest
+	QueryTaskOperation *pb.QueryTaskOperationRequest
+	QueryWorkerConfig  *pb.QueryWorkerConfigRequest
+
+	HandleSubTaskSQLs *pb.HandleSubTaskSQLsRequest
+	ExecDDL           *pb.ExecDDLRequest
+	BreakDDLLock      *pb.BreakDDLLockRequest
+
+	SwitchRelayMaster *pb.SwitchRelayMasterRequest
+	OperateRelay      *pb.OperateRelayRequest
+	PurgeRelay        *pb.PurgeRelayRequest
+	UpdateRelay       *pb.UpdateRelayRequest
+	MigrateRelay      *pb.MigrateRelayRequest
```

Comment:
listing all of these concrete types VS using an interface ([Message](https://godoc.org/github.com/golang/protobuf/proto#Message)), which is better?